tests/vcs/bitbucket-spec.js

```diff
@@ -0,0 +1,277 @@
+
+      resolution = rejection = null
+      promise = bitbucket.getPr('5')
+        .then((resp) => {
+          resolution = resp
+          return resolution
+        })
+        .catch((err) => {
+          rejection = err
+          throw err
+        })
+    })
+
+    it('should call fetch with proper params', function () {
+      expect(fetchStub).to.have.been.calledWith(`${bitbucket.baseUrl2}/repositories/owner/repo/pullrequests/5`)
+    })
+
+    describe('when fetch resolves with success', function () {
+      let resp
+      beforeEach(function (done) {
+        resp = {ok: true, status: 200, json () {}}
+        let pr = {
+          id: 5,
+          description: 'This is a #fix#',
+          source: {
+            commit: {
+              hash: 'sha-1'
+            }
+          },
+          links: {
+            self: {
+              href: 'my-link-to-myself'
+            }
+          }
+        }
+        sandbox.stub(resp, 'json').returns(Promise.resolve(pr))
+
+        promise.then(() => {
+          done()
+        })
+
+        fetchResolver.resolve(resp)
+      })
+
+      it('should resolve with the correct PR', function () {
+        expect(resolution).to.be.eql({
+          description: 'This is a #fix#',
+          headSha: 'sha-1',
+          number: 5,
+          url: 'my-link-to-myself'
+        })
+      })
+    })
+
+    describe('when fetch resolves with error', function () {
+      let resp, err
+      beforeEach(function (done) {
+        resp = {ok: false, status: 400, json () {}}
+        err = {
+          message: 'Uh oh'
+        }
+        sandbox.stub(resp, 'json').returns(Promise.resolve(err))
+
+        promise.catch(() => {
+          done()
+        })
+
+        fetchResolver.resolve(resp)
+      })
+
+      it('should not resolve', function () {
+        expect(resolution).to.equal(null)
+      })
+
+      it('should reject with the proper error', function () {
+        expect(rejection).to.be.eql(new Error(`400: ${JSON.stringify(err)}`))
+      })
+    })
+
+    describe('when fetch errors', function () {
+      beforeEach(function (done) {
+        promise.catch(() => {
+          done()
+        })
+
+        fetchResolver.reject('my-error')
+      })
+
+      it('should pass up the error', function () {
+        expect(rejection).to.be.equal('my-error')
+      })
+    })
+  })
+
+  describe('.postComment()', function () {
+    let resolution, rejection, promise, fetchResolver
+    beforeEach(function () {
+      fetchResolver = {}
+      let fetchPromise = new Promise((resolve, reject) => {
+        fetchResolver.resolve = resolve
+        fetchResolver.reject = reject
+      })
+
+      fetchStub.returns(fetchPromise)
+
+      resolution = rejection = null
+      promise = bitbucket.postComment('5', 'Missing PR scope!')
+        .then((resp) => {
+          resolution = resp
+          return resolution
+        })
+        .catch((err) => {
+          rejection = err
+          throw err
```

Comment:
Are you sure you want to re-throw here? I'm pretty sure that will make it hard to test this path, as the un-caught error will cause `mocha` to fail, at least, I think I've seen that behavior before.

Reply:
OH, I see, with the extra catch below it's not un-caught anymore.